### PR TITLE
[Tests-only] Check propfind response for various file/folder properties

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

This PR adds acceptance tests to check the response of various profind requests. This PR should cover the following properties:
```
    <d:getcontenttype />
    <d:getetag />
    <d:getlastmodified />
    <d:resourcetype />
    <oc:comments-unread />
    <oc:fileid />
    <oc:favorite />
    <oc:owner-display-name />
    <oc:permissions />
    <oc:size />
    <oc:tags />
```

Following properties have already been checked in the respective features
`<d:lockdiscovery />`: `tests/acceptance/features/apiWebdavLocks/publicLinkLockdiscovery.feature`
`<d:getcontentlength />`: `tests/acceptance/features/apiWebdavOperations/downloadFile.feature:88`
`<oc:share-types />`: `tests/acceptance/features/apiWebdavProperties/getFileProperties.feature`
## Related Issue

- https://github.com/owncloud/core/issues/36821

## How Has This Been Tested?

- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiTags/retreiveTags.feature [hunk 1]
```diff
@@ -15,6 +15,10 @@ Feature: tags
       | name       | type   |
       | MyFirstTag | normal |
     And the HTTP status when user "Brian" requests tags for file "/myFileToTag.txt" owned by user "Alice" should be "404"
+    When user "Alice" gets the following properties of file "/myFileToTag.txt" using the WebDAV API
+      | propertyName      |
+      | oc:tags           |
+    Then the single response should contain a property "oc:tags" with value ""
```
Comment (HariBhandari07) [inline tests/acceptance/features/apiTags/retreiveTags.feature:21]: @phil-davis when does `oc:tags` get value? Do you have any idea/suggestion ?

Comment (phil-davis) [reply tests/acceptance/features/apiTags/retreiveTags.feature:21]: I guess when a file is tagged?

Steps like in `tests/acceptance/features/apiTags/assignTags.feature` might be easy to use - create a tag, then add it to a file, then propfind on the file.

Comment (phil-davis) [reply tests/acceptance/features/apiTags/retreiveTags.feature:21]: That's sad - that is already in this scenario, but the tag is not in the propfind.

Comment (HariBhandari07) [reply tests/acceptance/features/apiTags/retreiveTags.feature:21]: @phil-davis This scenario also creates a tag, adds it to a file and propfind on the file. Still, `oc:tags`  is not present in response. Also, please have a look at the other scenarios added in this PR.

Comment (phil-davis) [reply tests/acceptance/features/apiTags/retreiveTags.feature:21]: We can merge this, and I will ask about how `oc:tags`  should get data in it.

File: tests/acceptance/features/apiWebdavProperties/getFileProperties.feature [hunk 2]
```diff
@@ -330,3 +330,159 @@ Feature: get file properties
...
       | /TestFolder/test2.txt | testprop2    | DDDDD                  |
       | /TestFolder/          | status       | HTTP/1.1 404 Not Found |
+
+  Scenario Outline: Propfind the last modified date of a folder using webdav api
+    Given using <dav_version> DAV path
+    And user "Alice" has created folder "/test"
+    When user "Alice" gets the following properties of folder "/test" using the WebDAV API
+      | propertyName      |
+      | d:getlastmodified |
+    Then the single response should contain a property "d:getlastmodified" with value like "/^[MTWFS][uedhfriatno]{2},\s(\d){2}\s[JFMAJSOND][anebrpyulgctov]{2}\s\d{4}\s\d{2}:\d{2}:\d{2} GMT$/"
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavProperties/getFileProperties.feature:340]: That's an impressive regex!

File: tests/acceptance/features/apiWebdavProperties/getFileProperties.feature [hunk 3]
```diff
@@ -330,3 +330,159 @@ Feature: get file properties
...
+    When user "Alice" gets the following properties of folder "file.txt" using the WebDAV API
+      | propertyName |
+      | oc:size      |
+    Then the single response should contain a property "oc:size" with value "16"
+    Examples:
+      | dav_version |
+      | old         |
+      | new         |
+
+  Scenario Outline: Propfind the size of a file using webdav api
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavProperties/getFileProperties.feature:406]:
```suggestion
  Scenario Outline: Propfind the size of a folder using webdav api
```

File: tests/acceptance/features/apiWebdavProperties/getFileProperties.feature [hunk 4]
```diff
@@ -330,3 +330,159 @@ Feature: get file properties
...
+    When user "Alice" gets the following properties of folder "/test" using the WebDAV API
+      | propertyName |
+      | oc:size      |
+    Then the single response should contain a property "oc:size" with value "0"
+    Examples:
+      | dav_version |
+      | old         |
+      | new         |
+
+  Scenario Outline: Propfind the file id a file using webdav api
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavProperties/getFileProperties.feature:418]:
```suggestion
  Scenario Outline: Propfind the file id of a file using webdav api
```

File: tests/acceptance/features/apiWebdavProperties/getFileProperties.feature [hunk 5]
```diff
@@ -330,3 +330,159 @@ Feature: get file properties
...
+    When user "Alice" gets the following properties of folder "file.txt" using the WebDAV API
+      | propertyName |
+      | oc:fileid    |
+    Then the single response should contain a property "oc:fileid" with value like '/[0-9]{10}/'
+    Examples:
+      | dav_version |
+      | old         |
+      | new         |
+
+  Scenario Outline: Propfind the file id a folder using webdav api
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavProperties/getFileProperties.feature:430]:
```suggestion
  Scenario Outline: Propfind the file id of a folder using webdav api
```

File: tests/acceptance/features/apiWebdavProperties/getFileProperties.feature [hunk 6]
```diff
@@ -330,3 +330,159 @@ Feature: get file properties
...
+    When user "Alice" gets the following properties of folder "/test" using the WebDAV API
+      | propertyName |
+      | oc:fileid    |
+    Then the single response should contain a property "oc:fileid" with value like '/[0-9]{10}/'
+    Examples:
+      | dav_version |
+      | old         |
+      | new         |
+
+  Scenario Outline: Propfind the owner display name a file using webdav api
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavProperties/getFileProperties.feature:442]:
```suggestion
  Scenario Outline: Propfind the owner display name of a file using webdav api
```

File: tests/acceptance/features/apiWebdavProperties/getFileProperties.feature [hunk 7]
```diff
@@ -330,3 +330,159 @@ Feature: get file properties
...
+    When user "Alice" gets the following properties of folder "file.txt" using the WebDAV API
+      | propertyName          |
+      | oc:owner-display-name |
+    Then the single response should contain a property "oc:owner-display-name" with value "Alice Hansen"
+    Examples:
+      | dav_version |
+      | old         |
+      | new         |
+
+  Scenario Outline: Propfind the owner display name a folder using webdav api
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavProperties/getFileProperties.feature:454]:
```suggestion
  Scenario Outline: Propfind the owner display name of a folder using webdav api
```

File: tests/acceptance/features/apiWebdavProperties/getFileProperties.feature [hunk 8]
```diff
@@ -330,3 +330,159 @@ Feature: get file properties
...
+    When user "Alice" gets the following properties of folder "file.txt" using the WebDAV API
+      | propertyName      |
+      | d:getetag  |
+    Then the single response should contain a property "d:getetag" with value like '%\"[a-z0-9]{32}\"%'
+    Examples:
+      | dav_version |
+      | old         |
+      | new         |
+
+  Scenario Outline: Propfind the resource type of a file using webdav api
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavProperties/getFileProperties.feature:382]: What happens when requesting `d:resourcetype` of a folder?
I don't see a scenario for that.

Comment (HariBhandari07) [reply tests/acceptance/features/apiWebdavProperties/getFileProperties.feature:382]: We already have scenarios for requesting `d:resourcetype` of a folder in `apiWebdavProperties/createFolder.feature` file : 
```
When user "Alice" gets the following properties of folder "/test_folder" using the WebDAV API
      | propertyName   |
      | d:resourcetype |
    Then the single response should contain a property "d:resourcetype" with a child property "d:collection"
```

Comment (phil-davis) [reply tests/acceptance/features/apiWebdavProperties/getFileProperties.feature:382]: For a file (this scenario) is there a child property?
"d:something" 

Or is `d:resourcetype` really empty?

**Review by phil-davis — APPROVED** [review]

File: tests/acceptance/features/apiWebdavProperties/getFileProperties.feature (hunk 8 above)
Comment (HariBhandari07) [reply tests/acceptance/features/apiWebdavProperties/getFileProperties.feature:382]: @phil-davis  for a folder the response is like: 
```
<d:resourcetype>
          <d:collection/>
</d:resourcetype>
```
 for a file it is empty:
```
<d:resourcetype/>
```

File: tests/acceptance/features/apiTags/retreiveTags.feature (hunk 1 above)
Comment (phil-davis) [reply tests/acceptance/features/apiTags/retreiveTags.feature:21]: I raised issue #37529

**Pull request merged**